pypeit/core/flat.py

```diff
@@ -1,488 +1,886 @@
+            = sorted_flat_data(norm_spec, spat_coo, gpm=_spat_gpm)
+    return _spat_gpm, spat_srt, spat_coo_data, spat_flat_data_raw, \
+                illum_filter(spat_flat_data_raw, med_width)
+
+
+def tweak_slit_edges(left, right, spat_coo, norm_flat, thresh=0.93, maxfrac=0.1):
+    r"""
+    Adjust slit edges based on the normalized slit illumination profile.
+
+    Args:
+        left (`numpy.ndarray`_):
+            Array with the left slit edge for a single slit. Shape is
+            :math:`(N_{\rm spec},)`.
+        right (`numpy.ndarray`_):
+            Array with the right slit edge for a single slit. Shape
+            is :math:`(N_{\rm spec},)`.
+        spat_coo (`numpy.ndarray`_):
+            Spatial pixel coordinates in fractions of the slit width
+            at each spectral row for the provided normalized flat
+            data. Coordinates are relative to the left edge (with the
+            left edge at 0.). Shape is :math:`(N_{\rm flat},)`.
+            Function assumes the coordinate array is sorted.
+        norm_flat (`numpy.ndarray`_)
+            Normalized flat data that provide the slit illumination
+            profile. Shape is :math:`(N_{\rm flat},)`.
+        thresh (:obj:`float`, optional):
+            Threshold of the normalized flat profile at which to
+            place the two slit edges.
+        maxfrac (:obj:`float`, optional):
+            The maximum fraction of the slit width that the slit edge
+            can be adjusted by this algorithm. If ``maxfrac = 0.1``,
+            this means the maximum change in the slit width (either
+            narrowing or broadening) is 20% (i.e., 10% for either
+            edge).
+
+    Returns:
+        tuple: Returns six objects:
+
+            - The threshold used to set the left edge
+            - The fraction of the slit that the left edge is shifted to
+              the right
+            - The adjusted left edge
+            - The threshold used to set the right edge
+            - The fraction of the slit that the right edge is shifted to
+              the left
+            - The adjusted right edge
+        
+    """
+    # Check input
+    nspec = len(left)
+    if len(right) != nspec:
+        msgs.error('Input left and right traces must have the same length!')
+
+    # Median slit width
+    slitwidth = np.median(right - left)
+
+    # ------------------------------------------------------------------
+    # Adjust the left edge
+
+    # Get the maximum to the left of the center
+    # TODO: Set a parameter for this
+    ileft = (spat_coo > 0.1) & (spat_coo < 0.4)
+    if not np.any(ileft):
+        msgs.error('No coordinates toward the left of the slit center.  Slit boundaries are '
+                   'likely in error, and you probably have a bad (very short) slit.  Slit center '
+                   'at center row is {0:.1f}.'.format((left[nspec//2] + right[nspec//2])/2))
+    left_thresh = thresh * np.amax(norm_flat[ileft])
```

Comment:
Yeah, this is what the previous code did.  Any comment @jhennawi ?

Reply:
The illumination profile is normalized to be close to one I believe, since we already took out the average spectral profile. It anyway only makes sense to set the cutoff in units of the maximum value of the profile. I don't see how else one would set this cutoff. An alternative option is to explicitly normalize the illumination profile to have a maximum value of unity before doing these steps, in which case threshold would be in units of that quantity.

Reply:
We talked through this.  Leaving it.